### DISCUSSION TITLE
updating /stats/request responses via a flag

### DISCUSSION THREAD

**Discussion by bijancot** [locustio/locust#2760] [Q&A]
Hi guys just wondering about /stats/requests url that from https://github.com/locustio/locust/blob/master/locust/stats.py module. can we add more responses in the json responses just using a flag instead of updating this responses 

```
        return {
            "method": escape(self.method or "") if escape_string_values else self.method,
            "name": escape(self.name) if escape_string_values else self.name,
            "safe_name": escape(self.name, quote=False),
            "num_requests": self.num_requests,
            "num_failures": self.num_failures,
            "avg_response_time": self.avg_response_time,
            "min_response_time": 0 if self.min_response_time is None else proper_round(self.min_response_time),
            "max_response_time": proper_round(self.max_response_time),
            "current_rps": self.current_rps,
            "current_fail_per_sec": self.current_fail_per_sec,
            "median_response_time": self.median_response_time,
            **response_time_percentiles,
            "avg_content_length": self.avg_content_length,
        }
```
since in that module also a dict StatsEntryDict exist?

Thankyou! i also post it in slack please let me know if is it not allowed to make duplicate discussion

**Comment by cyberw** [comment]
Hi! I'm sorry, but I don't really understand your question. What is your end goal, what have you tried, and why is that not working?

**Comment by bijancot** [comment]
Hi @cyberw thank you for the responses. And i'am sorry for the confusion. So it's working indeed but i want to change the responses of `/stats/requests` from :

```
{
  "current_response_time_percentiles": {
    "response_time_percentile_0.95": null
  },
  "errors": [],
  "fail_ratio": 0.0,
  "state": "ready",
  "stats": [
    {
      "avg_content_length": 0,
      "avg_response_time": 0.0,
      "current_fail_per_sec": 0,
      "current_rps": 0,
      "max_response_time": 0.0,
      "median_response_time": 0,
      "method": null,
      "min_response_time": 0,
      "name": "Aggregated",
      "num_failures": 0,
      "num_requests": 0,
      "response_time_percentile_0.95": 0,
      "response_time_percentile_0.99": 0,
      "safe_name": "Aggregated"
    }
  ],
  "total_avg_response_time": 0.0,
  "total_fail_per_sec": 0,
  "total_rps": 0,
  "user_count": 0
}
```

to add more results like 

```
{
  "current_response_time_percentiles": {
    "response_time_percentile_0.95": null
  },
  "errors": [],
  "fail_ratio": 0.0,
  "state": "ready",
  "stats": [
    {
      "avg_content_length": 0,
      "avg_response_time": 0.0,
      "current_fail_per_sec": 0,
      "current_rps": 0,
      "max_response_time": 0.0,
      "median_response_time": 0,
      "method": null,
      "min_response_time": 0,
      "name": "Aggregated",
      "num_failures": 0,
      "num_requests": 0,
      "response_time_percentile_0.95": 0,
      "response_time_percentile_0.99": 0,
      "safe_name": "Aggregated",
      "last_request_timestamp": float, 
      "start_time": float
    }
  ],
  "total_avg_response_time": 0.0,
  "total_fail_per_sec": 0,
  "total_rps": 0,
  "user_count": 0
}
```

i just want to modify the responses to add `start_time` and `last_request_timestamp` to `/stats/request` responses. So is there any way to modify these response by default by using some flags maybe since it's already mentioned in the struct that mentioned here https://github.com/locustio/locust/blob/master/locust/stats.py#L64-L76 ?

Thank you so much!
Best regards

**Comment by andrewbaldwin44** [comment]
Hi @bijancot, did you get an answer to your question yet? Perhaps you could provide some more information on your use case? If you want to record the timestamps for when the tests have started / ended, perhaps the [event hooks](https://docs.locust.io/en/stable/api.html#events) would suit your use case - particularly `test_start` and `test_stop`?
